ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/events/filesystem/DatabaseEventsIterator.java

```diff
@@ -170,7 +170,7 @@ public BootstrapEvent next() {
     }
 
     String currentPath = next.toString();
-    if (currentPath.contains(FUNCTIONS_ROOT_DIR_NAME)) {
+    if (currentPath.contains(Path.SEPARATOR + FUNCTIONS_ROOT_DIR_NAME)) {
```

Comment:
better to append Path.SEPARATOR after _functions for check

Reply:
I have added it both before and after now. However we are not allowed to create tables starting with _ in hive (like _functions_t1), gives a parse exception.